jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -88,6 +90,9 @@ YUI.add('juju-env-bakery', function(Y) {
             document.cookie = prefix + cfg.macaroon + ';path=/';
           }
         }
+        if (cfg.dischargeToken) {
+          window.localStorage.setItem(DISCHARGE_TOKEN, cfg.dischargeToken);
```

Comment:
Lets pass in the localStorage we want to use when creating the instance of bakery - that way we can easily pass in a localStorage stub for testing.